js/data/array_query.js

```diff
@@ -130,13 +136,16 @@ const defaultCompare = function(xValue, yValue) {
 };
 
 const SortIterator = Iterator.inherit({
-
     ctor: function(iter, getter, desc, compare) {
```

Comment:
Let's not override the ctor argument and change these lines a little:

```typescript
  ctor: function(iter, getter, desc, compare) {
    const { compareOptions } = iter;

    if(iter instanceof MapIterator) {
        this.iter = iter;
    } else {
        this.iter = new MapIterator(iter, this._wrap);
        this.iter.compareOptions = compareOptions;
    }

    // Question.
    // Do we need the compareOptions field in this class?
    // I didn't see any usage of this.compareOptions
    // But maybe I missed something :)
    this.compareOptions = compareOptions;
    this.rules = [{ getter: getter, desc: desc, compare: compare, compareOptions }];
}
```

Reply:
we need compareOptions  as props of current iterator